rskj-core/src/main/java/co/rsk/peg/pegininstructions/PeginInstructionsVersion1.java

```diff
@@ -0,0 +1,70 @@
+package co.rsk.peg.pegininstructions;
+
+import co.rsk.bitcoinj.core.Address;
+import co.rsk.bitcoinj.core.NetworkParameters;
+import org.ethereum.util.ByteUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public class PeginInstructionsVersion1 extends PeginInstructionsBase {
+    private static final Logger logger = LoggerFactory.getLogger(PeginInstructionsVersion1.class);
+    private static final int P2PKH_ADDRESS_TYPE = 1;
+    private static final int P2SH_ADDRESS_TYPE = 2;
+
+    private final NetworkParameters params;
+    private Optional<Address> btcRefundAddress;
+
+    public PeginInstructionsVersion1(NetworkParameters params) {
+        super(1);
+        this.params = params;
+    }
+
+    @Override
+    protected void validateDataLength(byte[] data) throws PeginInstructionsParseException {
+        if (data.length != 22 && data.length != 43) {
+            String message = String.format("[validateDataLength] Invalid data length. Expected 22 or 43 bytes, received %d", data.length);
+            logger.debug(message);
+            throw new PeginInstructionsParseException(message);
+        }
+    }
+
+    @Override
+    protected void parseAdditionalData(byte[] data) throws PeginInstructionsParseException {
+        if (data.length == 22) {
+            this.btcRefundAddress = Optional.empty();
+            return;
+        }
+
+        byte[] btcRefundAddressTypeBytes = Arrays.copyOfRange(data, 22, 23);
+        int btcRefundAddressType = ByteUtil.byteArrayToInt(btcRefundAddressTypeBytes);
+        byte[] hash = Arrays.copyOfRange(data, 23, data.length);
+
+        Address btcRefundAddress;
```

Comment:
```suggestion
        Address parsedBtcRefundAddress;
```
This will save you a code smell